source/common/router/upstream_request.cc

```diff
@@ -420,6 +420,11 @@ void UpstreamRequest::onPoolReady(
   stream_info_.setUpstreamSslConnection(info.downstreamSslConnection());
   parent_.callbacks()->streamInfo().setUpstreamSslConnection(info.downstreamSslConnection());
 
+  if (info.upstreamConnectionId().has_value()) {
+    stream_info_.setUpstreamConnectionId(info.upstreamConnectionId().value());
+    parent_.callbacks()->streamInfo().setUpstreamConnectionId(info.upstreamConnectionId().value());
```

Comment:
This only handles the http, should we handle the tcp case? I know the SSL connection info will be populated for tcpproxy also. https://github.com/envoyproxy/envoy/blob/32d40af143a0795a4fd45a5c0c4b59850e378543/source/common/tcp_proxy/tcp_proxy.cc#L531

Reply:
It's a good point - I was going to ask the same thing but it occurred to me that we might also want this for thrift and mysql and I think folks can add it as they want it.  the API says "not available" rather than "if there is no upstream connection" so I think it's simply not available yet for TCP

Reply:
+1 to adding exposure as desired.

Reply:
+1 to adding exposure as desired.

Reply:
> It's a good point - I was going to ask the same thing but it occurred to me that we might also want this for thrift and mysql and I think folks can add it as they want it. the API says "not available" rather than "if there is no upstream connection" so I think it's simply not available yet for TCP

got it, thanks, if the API says "not available", then we are cool here. I thought only the last filter is responsible for populating those stream info.(the tcp proxy and HCM). I thought MySQL and thrift also need the tcp proxy filter. But I may not understand correctly here.